### PR TITLE
fix: use correct commit and check type for deployments metric package_json

### PR DESCRIPTION
**Pull request by remmycat**

### CODE REVIEW

**Comment by remmycat** [comment]
Shortly tested that this is working using the dev command from the other branch.
See https://github.com/remmycat/telemetry-data-remmy/blob/main/raw-data/remmycat/webhook-trigger-playground/deployment/1702564533920.json (if you have access), the version correctly shows the deployed "1.0.1" version, but the main branch says "1.0.0".

File: src/metrics/deployments/index.ts [hunk 1]
```diff
@@ -90,7 +91,7 @@ export const collectDeploymentMetrics = async (
 
       packageJson = {
         is_parsable: true,
-        version: content.version,
+        version: typeof content.version === "string" ? content.version : null,
```
Comment (s2-julceb) [inline src/metrics/deployments/index.ts:94]: we should add a test that checks the unhappy path

**Review by s2-julceb — CHANGES_REQUESTED** [review]

**Comment by remmycat** [comment]
Sorry, I got a bit sidetracked with this.
- I added the tests
- Then I saw that it's called `is_parsable` here, but `parsable` in the tooling metric and aligned that
- Then I realised that the `exists` check should probably work the same too.
- I was duplicating a lot of code between these metrics, so I put most of the try/catch paths into a separate shared helper
- (This is where I briefly tried out the request-error approach, and then went back to the previous check)
- For now I covered all the cases in the metric's unit tests (so the coverage of the new helper is at 100% already), so I didn't add a unit test for the helper too.
- We arrived in the present time!

File: src/shared/log-messages.ts [hunk 2]
```diff
@@ -27,6 +27,11 @@ export enum LogWarnings {
   signingEventSignatureNotRecognized = "It is not possible to add a signature to the event.",
   collectMetricsSignatureNotRecognized = "It is not possible to collect metrics from this event.",
   documentationSkeletonConfigNotFound = "No documentation skeleton config file found. Owner/Repo is: '%s'",
+  documentationSkeletonConfigNotParsable = "Could not parse content of documentation skeleton config for owner/repo: '%s'",
+  documentationSkeletonConfigNonStringVersion = "Focumentation skeleton config version field does not contain string. Owner/Repo is: '%s'",
```
Comment (s2-julceb) [inline src/shared/log-messages.ts:31]: Typo
"Focumentation" -> "Documentation"
FYI I Use the [code spell checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) VS Code extension to catch stuff like that. Not sure though it if can differentiate between BE and AE.

Comment (remmycat) [reply src/shared/log-messages.ts:31]: What do you mean, that's what I call the documentation skeleton, it's a regional dialect from upstate New York…

Thanks for finding this :D I tried using the extension last year but couldn't deal so well with the constant noise from all the false positives.

**Review by s2-julceb — APPROVED** [review]

**Pull request merged**